hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/NetUtils.java

```diff
@@ -589,7 +589,7 @@ public static void connect(Socket socket,
     } catch (SocketTimeoutException ste) {
       throw new ConnectTimeoutException(ste.getMessage());
     }  catch (UnresolvedAddressException uae) {
-      throw new UnknownHostException(uae.getMessage());
+      throw new UnknownHostException(endpoint.toString());
```

Comment:
we still need that message from the original string so build it from `uhe.getMessage() + " " + endpoint`

Reply:
Thank you for taking a look. I considered this, but a UHE cannot have a message. It only as has no args constructor so it cannot be created with a message. There's also no setMessage method as far as I can tell, so can't be added after. https://docs.oracle.com/javase/8/docs/api/java/net/UnknownHostException.html